src/App.php

```diff
@@ -302,32 +306,87 @@ public function outputDebug($str)
         echo 'DEBUG:' . $str . '<br/>';
     }
 
+    /**
+     * Normalize HTTP headers to associative array with LC keys.
+     *
+     * @param string[] $headers
+     * @return string[]
+     */
+    protected function normalizeHeaders(array $headers): array
+    {
+        $res = [];
+        foreach ($headers as $k => $v) {
+            if (is_numeric($k) && ($p = strpos($v, ':')) !== false) {
+                $k = substr($v, 0, $p);
+                $v = substr($v, $p + 1);
+            }
+
+            $res[strtolower(trim($k))] = trim($v);
+        }
+
+        return $res;
+    }
+
+    /**
+     * @return $this
```

Comment:
no comment what this method does.

Reply:
I am very strictly against any duplication/copying of the same - writing there "set Response Status Code" from the method header "setResponseStatusCode" is dangerous when the method is refactored, it takes only real-estate in the code and does provide zero value. So no single advantage.

What is the purpose of it?

Reply:
About duplicating you're definitely right. But comments is useful when you check code and to easily understand what each method does. Normally it would be very good doc where all methods are described, but toolkit don't have such extensive documentation, so quite often I find myself reading toolkit code instead to understand how it works :)